packages/hardhat-core/src/internal/core/jsonrpc/types/input/subscribeRequest.ts

```diff
@@ -0,0 +1,21 @@
+import * as t from "io-ts";
+
+import { RpcFilterRequest } from "./filterRequest";
+
+// TODO: This types are incorrect. See: https://geth.ethereum.org/docs/rpc/pubsub
```

Comment:
This is very confusing. I even suspect that Geth's documentation is wrong, and our implementation is correct. I prefer to leave a TODO for now.